hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/HoodieInputFormat.java

```diff
@@ -62,65 +66,53 @@
 
   @Override
   public FileStatus[] listStatus(JobConf job) throws IOException {
-    // Get all the file status from FileInputFormat and then do the filter
-    FileStatus[] fileStatuses = super.listStatus(job);
-    Map<HoodieTableMetaClient, List<FileStatus>> groupedFileStatus = groupFileStatus(fileStatuses);
-    LOG.info("Found a total of " + groupedFileStatus.size() + " groups");
+    // Segregate inputPaths[] to incremental, non incremental and non hoodie paths
+    List<String> incrementalTables = HoodieHiveUtil.getIncrementalTableNames(Job.getInstance(job));
+    InputPathHandler inputPathHandler = new InputPathHandler(getInputPaths(job), incrementalTables);
     List<FileStatus> returns = new ArrayList<>();
-    for (Map.Entry<HoodieTableMetaClient, List<FileStatus>> entry : groupedFileStatus.entrySet()) {
-      HoodieTableMetaClient metadata = entry.getKey();
-      if (metadata == null) {
-        // Add all the paths which are not hoodie specific
-        returns.addAll(entry.getValue());
+
+    Map<String, HoodieTableMetaClient> tableMetaClientMap = inputPathHandler.tableMetaClientMap;
+    // process incremental pulls first
+    for (String table : incrementalTables) {
+      HoodieTableMetaClient metaClient = tableMetaClientMap.get(table);
+      if (metaClient == null) {
```

Comment:
No. This is for incremental tables with no input paths.  Since we are taking incremental tables list from configs, its possible (rare corner case) that a table was set as incremental accidentally but no input paths were set for same.

Reply:
I added a comment to explain this.

Reply:
Could you please give an example of when this happens ?

Reply:
^

Reply:
This is possible when a Hive session property is set for a Hoodie table to consume in INCREMENTAL mode but then that table is never queried ( in which case no input path corresponds to that table)